enable sysinfo support on illumos systems

jclulow opened the pull request:
Support for illumos systems was added to the sys-info crate some time
ago, and was working with effective-limits, but as of 887a5a06f62
appears to have effectively been turned off at build time.

With this change, I can `cargo build` and `cargo test` without failures:

```
 $ cargo test
   Compiling effective-limits v0.5.3-alpha.0 (/ws/safari/effective-limits.rs)
    Finished test [unoptimized + debuginfo] target(s) in 0.41s
     Running target/debug/deps/effective_limits-91122e1efdd826d7

running 2 tests
test tests::test_min_opt ... ok
test tests::test_ulimit ... ok

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/test_limited-060e6d3400d5471b

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

   Doc-tests effective-limits

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```

I expect this will make the `error: 'sysinfo not supported on this platform'` message emitted from `rustup` on illumos systems go away.

jclulow commented:
I'm not really sure what to do about these CI/CD failures.  They don't seem related to the change?

jclulow commented:
Hi, @rbtcollins!  Do you think you could take a look at this one?  I'm not sure what's happening with the CI/CD checks, but I'm happy to do any additional manual builds or verification you'd like prior to integration.  Thanks!

rbtcollins commented:
Looks like Github Actions made a backwards incompatible change due to a security issue, which I hadn't noticed due to no PRs that I was looking at (and I missed yours - sorry, bad SnR in my inbox). Thanks for the ping!

The run configuration in this repo is closely modelled on rustups; rustup probably has a change from october 2020 to address the same thing, that may even be diff | patch appliable to this repo to fix the actions errors.

rbtcollins commented:
Ok, @kinnison has fixed the actions; can you rebase on master?

jclulow commented:
I have rebased and fixed up the change a bit to fit in with what's gone on in the meantime.  Build and test on an illumos system still passes:

```
$ cargo test
    Finished test [unoptimized + debuginfo] target(s) in 0.02s
     Running target/debug/deps/effective_limits-765ae6e07e287690

running 4 tests
test tests::test_min_opt ... ok
test tests::it_works ... ok
test tests::test_ulimit ... ok
test tests::test_no_ulimit ... ok

test result: ok. 4 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.01s

     Running target/debug/deps/test_limited-028622ff788761c2

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

   Doc-tests effective-limits

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
```

rbtcollins commented:
Thank you
